File: leetcode/easy/decode-the-message.ts
// https://leetcode.com/problems/decode-the-message/

// MINE
function decodeMessage(key: string, message: string): string {
  const map = new Map()
  let code = 97,
    ans = ''
  for (let c of key) if (c !== ' ' && !map.has(c)) map.set(c, String.fromCharCode(code++))
  for (let c of message) ans += map.get(c) || ' '
  return ans
}

File: leetcode/easy/decode-the-message.py
# https://leetcode.com/problems/decode-the-message/

class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        mp, i = {' ': ' '}, 97
        for c in key:
            if c != ' ' and c not in mp:
                mp[c] = chr(i)
                i += 1
        return ''.join(mp[c] for c in message)
